File: Problems/Algorithms/997. Find the Town Judge/FindTownJudge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] scores = new int[n+1];

        for (int i = 0; i < trust.length; i++) {
            scores[trust[i][0]]--;
            scores[trust[i][1]]++;
        }

        int i = 1;
        for (; i <= n; i++) {
            if (scores[i] == n - 1) {
                break;
            }
        }

        return i <= n ? i : -1;
    }
}

File: Problems/Algorithms/997. Find the Town Judge/find_town_judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        scores = [0 for _ in range(n+1)]

        for pair in trust:
            scores[pair[0]] -= 1
            scores[pair[1]] += 1

        i = 1
        while i < n+1:
            if scores[i] == n-1:
                break
            i += 1

        return i if i <= n else -1

